Prompt to reuse or ignore an existing GIT_HUB_MSG_FILE

perlpunk opened the pull request:
This needs a bit more work.

See issue #28

After a successful command, delete GIT_HUB_MSG_FILE.
When starting editor and file exists, prompt for reusing, deleting
or aborting.

Still todo:

- [x] Only delete the file after commands that actually use
the editor, so a non-editor-command wouldn't destroy an old file.
- [x] Delete when nothing was entered.
- [x] Show the age of the old file when prompting.
- [x] If reusing, replace the meta information with the one from the new command.

perlpunk commented:
Implemented the missing todos.
I squashed everything to a single commit.

ingydotnet commented:
sweet!

perlpunk commented:
:D
I forgot to remove a comment line. pushed force

perlpunk commented:
btw: it will ignore old msg files, so when users upgrade, they won't get prompted because of old files from successful commands.

ingydotnet commented:
Is there a way to detect and delete the old ones?

perlpunk commented:
I think it's not necessary to delete them.
But if you want, you can check for "# Command: " in the last line of the file (like it is done in `init-msg-file`), which is only there with the new code.

perlpunk commented:
another thing: when prompted, you can enter a, r or d. if the user enters something else, what should be done? abort?

perlpunk commented:
You were right, the old content was shown when there was an old
file. fixed.
Also, if the users enters something else than 'r' or 'd', it behaves
like 'a' (abort)

squashed commits

ingydotnet commented:
Here is a test where I entered `"foo\nbar\n"` as my msg:

```
~/src/test-repo $ git hub issue

Error: malformed message in '.git/GIT_HUB_EDIT_MSG'

  Usage: git hub issue-new [<owner>/<repo>]

  Create a new issue for a repository.

~/src/test-repo $ git hub issue
usage: date [-jnu] [-d dst] [-r seconds] [-t west] [-v[+|-]val[ymwdHMS]] ...
            [-f fmt date | [[[mm]dd]HH]MM[[cc]yy][.ss]] [+format]

.git/GIT_HUB_EDIT_MSG still exists, previous command probably wasn't successful.
The command was: 'git hub issue-new '
Date:
You can reuse or delete the old file.
(a)bort, (r)euse or (d)elete? a
~/src/test-repo $
```

The date failure is likely OSX specific. There are some OSX specific parts in
git-hub. Search for 'darwin'. Solve this in a similar fashion.

The command has an extra space at the end.

I'd like the prompt message to be:

```
Found existing GIT_HUB_EDIT_MSG from:

  command: '...'
  date:    ...

(r)euse, (s)how, (n)ew, (d)elete, (a)bort? 
```

Where 'show' runs something like `head .git/GIT_HUB_EDIT_MSG | grep -v '^#'`

And new opens a new message without deleting the old GIT_HUB_EDIT_MSG.

perlpunk commented:
- changed the prompt message format.
- added (s)how. it will prompt again after that

TODO:
- I cannot figure out how to call `date` under osx. Well I have an idea with using `stat` but I cannot test
- Opening a new file without deleting/overwriting the old one would require more changes because I'd have to use a different filename

perlpunk commented:
fixed the trailing whitespace

perlpunk commented:
Regarding choosing "new":
Choosing a new filename for the current file would make things more complicated. And it would get worse if the current command also fails, and so on.

I'd rather have the option to (s)ave the old file somewhere else, but this must be a unique filename in case this happens more than once. Like `file.<timestamp>`, or let the user choose a filename.

perlpunk commented:
I added an OSX specific command for getting the date of the file.
please test =)

ingydotnet commented:
Applied. Pushed. Thanks!

ingydotnet commented:
Skipping (n)ew option or saveas for now.
